packages/nibijs/src/result.ts

```diff
@@ -0,0 +1,86 @@
+/** Poor-man's Result type from Rust.
+ *
+ * The Result type forces you to explicitly handle erros in contrast to allowing
+ * errors to propagate up the call stack implicitly. Handling potential errors
+ * explicitly leads to more robust and reliable code.
+ *
+ * Ref: <a href="https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#propagating-errors">Propagating Errors - Rust Book</a>.
+ *
+ * @example
+ * // ---------------------------------------
+ * // Most common use-case: Result.ofSafeExec
+ * // ---------------------------------------
+ * res = Result.ofSafeExec(somethingDangerous) // without args
+ *
+ * // with args
+ * res = Result.ofSafeExec(() => somethingDangerous(arg0, arg1))
+ *
+ * @example
+ * // ---------------------------------------
+ * // Direct constructor
+ * // ---------------------------------------
+ * let res = new Result({ ok: "Operation successful!" })
+ * if (res.isOk()) {
+ *   happyPath(res.ok)
+ * } else {
+ *   handleGracefully(res.err!) // throws impossible based on constructor args
+ * }
+ *
+ * */
+export class Result<T> {
+  ok: T | undefined
+  err: Error | undefined
+  constructor({ ok, err }: { ok?: T; err?: unknown }) {
+    this.ok = ok
+    this.err = err ? parseError(err) : undefined
+
+    const undefinedArgsCount =
+      +(this.ok !== undefined) + +(this.err !== undefined)
```

Comment:
This line is confusing, initially, make it more readable or add a comment

Reply:
An ex.

const undefinedArgsCount = Boolean(this.ok) || Boolean(this.err) ? 1 : 0;

Reply:
Simplified to 
```ts
    if (Boolean(this.ok) && Boolean(this.err)) {

```